include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -1004,8 +1012,8 @@ void mergeSubproblems(SizeType i_begin, SizeType i_split, SizeType i_end, WorkSp
   // - reorder columns in `mat_ev` using `i2` such that eigenvectors match eigenvalues
   //
   sortIndex(i_begin, i_end, k_fut, ws.dtmp, ws.i1, ws.i2);
-  applyIndex(i_begin, i_end, ws.i2, ws.dtmp, d);
-  dlaf::permutations::permute<Backend::MC, Device::CPU, T, Coord::Col>(i_begin, i_end, ws.i2, ws.mat1,
-                                                                       mat_ev);
+  applyIndex(i_begin, i_end, ws.i2, ws.dtmp, ws_h.evals.get());
+  dlaf::permutations::permute<backend, device, T, Coord::Col>(i_begin, i_end, ws.i2, ws.mat1, evecs);
```

Comment:
Would it be useful to have `Matrix<T, Source> MatrixMirror::get_source()` so that you don't need to pass the original `evecs` into this function?

Reply:
Yes, I think it will be useful!

Reply:
See https://github.com/eth-cscs/DLA-Future/pull/620. Does not have to be included in this PR.

Reply:
@teonnik note that #620 was merged so you should be able to use `getSource` now. However, I think the code has changed enough in the meantime that I can't quite tell if you need it anymore...